packages/terra-framework-docs/src/terra-dev-site/doc/popup/example/PopupWithArrow.jsx

```diff
@@ -18,25 +19,66 @@ function PopupArrow() {
 
   const handleRequestClose = () => {
     setOpen(false);
+    buttonElement.current.focus(); // Return focus to the button when the popup closes
   };
 
+  const handleKeyDown = (event) => {
+    if (event.key === 'ArrowDown') {
+      event.preventDefault();
+      setFocusedIndex((prevIndex) => (prevIndex + 1) % 2);
+    } else if (event.key === 'ArrowUp') {
+      event.preventDefault();
+      setFocusedIndex((prevIndex) => (prevIndex - 1 + 2) % 2);
+    }
+  };
+
+  useEffect(() => {
+    if (open) {
+      setFocusedIndex(0);
```

Comment:
instead of initialising it on useEffect we can reset it to 0 on `handleRequestClose` method

Reply:
While implementing the same ,with keyboard navigation user required to press after opening the toggle dialog to move between the links.